development/tools/cmd/synchronizer/syncomponent/component.go

```diff
@@ -0,0 +1,104 @@
+package syncomponent
+
+import (
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const defaultOutOfDateDays = 3
+
+// Component represents the element for which version will be checked
+type Component struct {
+	Name       string
+	Path       string
+	GitHash    string
+	CommitDate string
+	outOfDate  bool
+	Versions   []*ComponentVersions
+}
+
+// ComponentVersions is a part of Component which includes information about version
+type ComponentVersions struct {
+	VersionPath   string
+	Version       string
+	ModifiedFiles []string
+	outOfDate     bool
+}
+
+// NewSynComponent returns new Component element
+func NewSynComponent(componentPath string, versionPaths []string) *Component {
+	componentVersions := []*ComponentVersions{}
+	for _, versionPath := range versionPaths {
+		componentVersion := &ComponentVersions{VersionPath: versionPath}
+		componentVersions = append(componentVersions, componentVersion)
+	}
+
+	nameElements := strings.Split(componentPath, "/")
+	nameDash := nameElements[len(nameElements)-1]
+	name := strings.Replace(nameDash, "-", "_", -1)
+
+	return &Component{
+		Name:     name,
+		Path:     componentPath,
+		Versions: componentVersions,
+	}
+}
+
+// SetOutOfDate determines whether a given component is not expired
+func (c *Component) SetOutOfDate(outOfDateDays int) {
```

Comment:
name of this method does not indicate what it is doing. I would expect that just copy value provided as an argument to a component field, but in reality, it does a lot more.

Reply:
👍